39 feature/flashcards

Kamil-Drozdz opened the pull request.

Kamil-Drozdz committed:
refactor:
correcting the application roots, removing 'use client' from them, wrapping and moving components to be used on the client side, adding missing arrow functions for conventions, improving the styling of registration, login and password reminder, removing redundant code from redux - currently not applicable

Kamil-Drozdz committed:
refactor: review fixes for comments

Kamil-Drozdz committed:
refactor: review fix, removing unnecessary mockData and placing them from the appropriate folder

Kamil-Drozdz committed:
feature:
adding a new initial functionality - flashcards, adding mock fetching on the SSR side as well as improving a few styling to several subpages

Kamil-Drozdz committed:
Resolved merge conflicts

xyashino commented on src/app/(UserContent)/layout.tsx:
```diff
@@ -1,8 +1,8 @@
-import UserContentNavigation from "@/components/wrappers/UserContentNavigation";
+import UserContentNavigation from '@/components/wrappers/UserContentNavigation';
 
 export default function UserContentLayout({ children }: { children: React.ReactNode }) {
```
There is a type in React called `PropsWithChildren`, and you can write it instead of `{ children: React.ReactNode }`

Kamil-Drozdz replied:
nice note

xyashino commented on src/app/(UserContent)/home/[category]/[subcategory]/[level]/[content]/page.tsx:
Styles need to be changed to the size of the tailwinds (e.g left-4) , let's avoid hardcoding values.

Kamil-Drozdz replied:
we'll play around with it once we have a brief on the final look at the moment most things are... just :P

xyashino commented on src/components/Flashcards.tsx:
```diff
@@ -0,0 +1,119 @@
+	flashcards: Flashcard[];
+};
+
+type FlashcardsProps = {
```
Change it to the interface

Kamil-Drozdz replied:
done

xyashino commented on src/components/Flashcards.tsx:
The types 'Flashcard' and 'Category' should be interfaces in the types directory.

xyashino replied:
I recommend moving the entire logic about flash cards to the useFlashcards hook as it will improve readability

Kamil-Drozdz replied:
done, we can go either with the technique which seems to be more difficult in fact the fact is reusable to some extent, however, in this case there will be no need for reusability the issue I am talking about is called Container/Presentation, where the logic is closed in the container and the logic is sent in props to presentation

xyashino commented on src/components/HomeContent/quizContent/QuizContainer.tsx:
Task and Quiz should be in the types directory

Kamil-Drozdz replied:
done

Kamil-Drozdz committed:
fix:
adding a custom hook that stores logic to Flashcards, changing ReactNode to PorpsWithChildren, changing type to interface for uniformity
